feat: add deposit and withdraw functionality via Boltz in wallets' view

krrprr opened the pull request:
Closes https://github.com/ExchangeUnion/xud-ui/issues/37

krrprr committed:
feat: add deposit and withdraw functionality via Boltz in wallets' view

erkarl reviewed (commented):
utACK

krrprr commented on src/api.ts:
```diff
@@ -26,9 +32,20 @@ const logAndThrow = (url: string, err: string) => {
+      }
+      return resp.json().then((err) => {
+        throw err;
+      });
```
:+1: thanks for the hint! Fixed.

erkarl reviewed (approved):
Approve code wise. Needs manual test by @raladev

raladev commented:
- [x] no one thinks in satoshi, we need to convert all satoshi into user-readable values for desktop users
[Screenshot from 2020-11-25 16-29-19](https://user-images.githubusercontent.com/29906866/100236447-bec52100-2f3e-11eb-9c0f-5c786962fca0.png)
- [x] withdrawal and deposit limits are different (withdrawal is lower than deposit), looks pretty strange. Also withdrawal range does not depend of my balance, it is too not good :(
[Screenshot from 2020-11-25 16-29-25](https://user-images.githubusercontent.com/29906866/100236479-c84e8900-2f3e-11eb-8b71-b9f4a73e0af7.png)
- [x] window jumping when clilcking on deposit/withdraw (or Trading limits/back) button looks annoying. can we set minimal size of the box, that will prevent such jumping?
- [x] `Check detailed status in the console tab` looks like a button, but it is not. Would be good to make button from this string -> after clicking console view will be opened + string for status lookup will be printed in console. (becaouse otherwise user will have no idea how to get ID of current transaction to get the status)
- [x] qr in withdrawal section looks redundant. I dont see any cases where it can be useful.
[Screenshot from 2020-11-25 17-13-46](https://user-images.githubusercontent.com/29906866/100238880-c2a67280-2f41-11eb-8276-a23d7decdc55.png)
- [x] Would be good to map such errors to something  user-friendly:
[Screenshot from 2020-11-25 17-25-54](https://user-images.githubusercontent.com/29906866/100240301-778d5f00-2f43-11eb-8984-1c21814299a4.png)
[Screenshot from 2020-11-25 17-27-35](https://user-images.githubusercontent.com/29906866/100240358-8bd15c00-2f43-11eb-9421-a625392408cc.png)

-  We definitely need to think about explanation and presentation of boltz deposits and withdrawals in ui: 

a) Also for me it looks pretty strange that after deposit im getting channel with balance, it is not default behavior for lightning based solutions, and users may be confused, because on deposit page we have on-chain balance and off-chain balance.  

b)After deposit im getting off-chain balance that is fully available for the trade, but what if i dont want to trade for a full amount?

c) Also there is no mention about remote balance that i will get from boltz, but we have notification about remote balance - will look strange too.

d) I should pay 1 % in decentralized exchange for deposit and for withdrawal... Maybe we should add some (i) button with  explanation why i need to pay 1% and what i will get?

e) If i got on-chain balance (for example after force-closing) and will try to withdraw it to my external wallet using ui i will got inssuf funds err, because withdrawal works only for boltz channels. (also i will get same err if will try to withdraw from non-boltz channels)

kilrau commented:
Agreed on call with @krrprr @raladev we will handle all of the above (and more) in a follow-up issue: https://github.com/ExchangeUnion/xud-ui/issues/118

raladev commented:
- [ ] it would be good to hide deposit and withdraw buttons if status does not return boltz container (simnet case), and make this buttons unavailable if boltz status is red (disabled, exited, e.t.c.)
![Screenshot from 2020-11-25 19-51-36](https://user-images.githubusercontent.com/29906866/100258353-0c9a5300-2f58-11eb-8421-974c7a6f15b4.png)

kilrau commented:
Yes, agree. Best way would be to grey the buttons out with hover tooltip saying `Not available on Simnet` (Simnet) and `Boltz Error` (Testnet/Mainnet).

krrprr commented:
Implemented disabling the `Deposit` and `Withdraw` buttons in case of Simnet, Boltz status not ready, and Boltz status error. Added tooltip with additional message when the buttons are disabled.
